### PR TITLE
storekeys: get more data from query

### PR DESCRIPTION
**Pull request by Olias**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,23 +34,25 @@ Create an `<auto-complete>` element with a child `input` to use as the data-entr
...
+| `resultdata` | the name of the property containing a result array of objects in the returned API JSON (not required if only results are returned)    |
+| `resultname` | the name of the property in each result object which matches the search input and is used for datalist `<option>` elements (optional) |
+|   `querymin` | the minimum number of characters to enter before a search occurs (default: 1)                                                         |
+| `inputdelay` | the minimum time to wait in milliseconds between keypresses before a search occurs (default debounce: 300)                            |
+|  `optionmax` | the maximum number of auto-complete options to show (default: 20)                                                                     |
+|      `valid` | if set, this error message is shown when an invalid value is selected                                                                 |
+|  `storekeys` | if set, these datadfields are addet as custom attributes in the \<options\> of the datalist                                           |
```
Comment (craigbuckler) [inline README.md:55]: Small typos: `these datafields are added`

Perhaps better:
if set to a comma-delimited list of result values, e.g. `storekeys="id,name"`, the `<options>` element has associated `data-` attributes set which can be accessed by the [`dataset` property](https://developer.mozilla.org/docs/Web/API/HTMLElement/dataset).

File: src/datalist-ajax.js [hunk 2]
```diff
@@ -186,10 +186,22 @@ class AutoComplete extends HTMLElement {
...
           const value = this.resultname ? data[d][this.resultname] : data[d];
+          const res = [];
+          if (this.storekeys){
+              let keys = this.storekeys.split(',');
+              for (let k of keys) {
+                  if (k in data[d]){
+                    res[k] = data[d][k];
+                  }
+              }
+          }
```
Comment (craigbuckler) [inline src/datalist-ajax.js:197]: Will this work for nested properties that you support elsewhere, e.g. `storekeys="name.first, name.last"`?

File: src/datalist-ajax.js [hunk 3]
```diff
@@ -186,10 +186,22 @@ class AutoComplete extends HTMLElement {
         for (let d = 0; d < data.length && optMax > 0; d++) {
 
           const value = this.resultname ? data[d][this.resultname] : data[d];
+          const res = [];
+          if (this.storekeys){
+              let keys = this.storekeys.split(',');
```
Comment (craigbuckler) [inline src/datalist-ajax.js:191]: You may need to `trim()` the values so `storekeys="id, name, age"` works as expected.

**Review by craigbuckler — CHANGES_REQUESTED** [review]
Nice idea! A few minor comments.

File: src/datalist-ajax.js (hunk 2 above)
Comment (Olias) [reply src/datalist-ajax.js:198]: ive change this, now it works,

File: README.md (hunk 1 above)
Comment (Olias) [reply README.md]: i didn't know this feature - cool

**Comment by Olias** [comment]
you are right... !

**Pull request merged**